src/NServiceBus.Core/Pipeline/Incoming/LoadHandlersConnector.cs

```diff
@@ -57,8 +66,29 @@ async Task<CompletableSynchronizedStorageSession> AdaptOrOpenNewSynchronizedStor
                    ?? await synchronizedStorage.OpenSession(contextBag).ConfigureAwait(false);
         }
 
+        static void LogHandlersInvocation(IIncomingLogicalMessageContext context, List<MessageHandler> handlersToInvoke)
+        {
+            var builder = new StringBuilder();
```

Comment:
do we really want to use a multi-line log entry? Otherwise we could easily remove the stringbuilder usage.

Reply:
I was considering having a sinle line entry, but then looking at amount of information added by headers _and_ the original [issue discussion](https://github.com/Particular/NServiceBus/issues/5179#issuecomment-393431440) which framed the expected and agreed upon format, didn't feel that would be a good idea.